content/blog/2021-01-26-january-21-community-gems.md

```diff
@@ -0,0 +1,189 @@
+  compressed datasets, and DVC pipelines in CI/CD.
+descriptionLong: |
+  A roundup of technical Q&A's from the DVC community. 
+  This month: parallelize your data transfer, 
+  compressed datasets, and DVC pipelines in CI/CD.
+picture: 2021-01-28/gems-cover.png
+author: elle_obrien
+commentsUrl: https://discuss.dvc.org/t/january-21-community-gems/645
+tags:
+  - Discord
+  - Gems
+  - CML
+  - Plots
+  - Pipelines
+  - Docker
+---
+
+## DVC questions
+
+### [Q: Is there an equivalent of `git restore <file>` for DVC?](https://discord.com/channels/485586884165107732/563406153334128681/799598181310267392)
+
+Yes! You'll want `dvc checkout`. It restores the corresponding verion of your
+DVC-tracked file or directory from
+[the cache](https://dvc.org/doc/user-guide/dvc-internals#structure-of-the-cache-directory)
+to your local workspace.
+[Read up in our docs for more info!](https://dvc.org/doc/command-reference/checkout#checkout)
+
+### [Q: My dataset is made of more than _a million_ small files. Can I use an archive format, like `tar.gz` with DVC?](https://discord.com/channels/485586884165107732/485596304961962003/798983422965841920)
+
+There are some downsides to using archive formats, and often we discourage it-
+but let's review some factors to consider, so you can make the best choice for
+your project.
+
+- If your `tar.gz` file changes at all- perhaps because you changed a single
+  file before zipping- you'll end up with an entirely new copy of the archive
+  every time you commit! This is not very space efficient, but if space isn't an
+  issue it might not be a dealbreaker.
+- Because of the way we optimize data transfer, you'll end up transferring the
+  whole archive anytime you modify a single file and `dvc push`/`dvc pull`.
+- In general, archives don't play nice with the concept of diffs. Looking back
+  at your git history, it can be challenging to log how files were deleted,
+  modified, or added when you're versioning archives.
+
+While we can't do much about the general issues that archives present for
+version control systems, DVC does have some options that might help you achieve
+better data transfer speeds. We recommend exploring DVC's built-in parallelism-
+data transfer functions like `dvc push` and `dvc pull` have a flag (`-j`) for
+increasing the number of jobs run simultaneously.
+[Check out the docs for more details](https://dvc.org/doc/command-reference/push#options).
+
+In summary, the advantage of using an archive format will depend on both how
+often you modify your dataset and how often you need to push and pull data. You
+might consider exploring both approaches (with and without compression) and run
+some speed tests for your use case. We'd love to know what you find!
+
+### [Q: My DVC remote is a server with a self-signed certificate. When I push data, DVC is giving me an SSL verification error- how can I get around this?](https://discord.com/channels/485586884165107732/563406153334128681/800707271502856222)
```

Comment:
It looks like the question is more for s3-compatible storage. Even for HTTP storage, can we suggest the `ssl_verify` at the end? It should be a last resort solution as it's insecure. There's `REQUESTS_CA_BUNDLE` env var similar to `AWS_CA_BUNDLE` that could be used (though I am not sure how well it works with dvc).

Reply:
@skshetry sounds good! could you please do a minor PR to improve this? :)